[new release] csexp (1.3.0)

jeremiedimino opened the pull request:
Parsing and printing of S-expressions in Canonical form

- Project page: <a href="https://github.com/ocaml-dune/csexp">https://github.com/ocaml-dune/csexp</a>
- Documentation: <a href="https://ocaml-dune.github.io/csexp/">https://ocaml-dune.github.io/csexp/</a>

##### CHANGES:

- Add a "feed" API for parsing. This new API let the user feed
  characters one by one to the parser. It gives more control to the
  user and the handling of IO errors is simpler and more
  explicit. Finally, it allocates less (ocaml-dune/csexp#9, @jeremiedimino)

- Fixes `input_opt`; it was could never return [None] (ocaml-dune/csexp#9, fixes ocaml-dune/csexp#7,
  @jeremiedimino)

- Fixes `parse_many`; it was returning s-expressions in the wrong
  order (ocaml-dune/csexp#10, @rgrinberg)

camelus commented:
Commit: 1c112f3a1d1ee1d8338f804ddd91c0b7b2c52a9e

A pull request by opam-seasoned @jeremiedimino.

##### :sunny: All lint checks passed <small>1c112f3a1d1ee1d8338f804ddd91c0b7b2c52a9e</small>

* These packages passed lint tests: `csexp.1.3.0`


---


##### :sunny: Installability check (+1)

* new installable packages (1): `csexp.1.3.0`

rgrinberg commented:
There's an issue in 4.02:

```
#=== ERROR while compiling csexp.1.3.0 ========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(https://github.com/ocaml-dune/csexp/releases/download/1.3.0/csexp-1.3.0.tbz)
# path                 ~/.opam/4.02/.opam-switch/build/csexp.1.3.0
# command              ~/.opam/4.02/bin/dune build -p csexp -j 72 @install
# exit-code            1
# env-file             ~/.opam/log/csexp-23-462d2b.env
# output-file          ~/.opam/log/csexp-23-462d2b.out
### output ###
#     ocamldep src/.csexp.objs/csexp.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamldep.opt -modules -impl src/csexp.ml) > _build/default/src/.csexp.objs/csexp.ml.d
# File "src/csexp.ml", line 143, characters 2-607:
# Error: Syntax error: attributes not expected.
```

jeremiedimino commented:
Ah, I disabled the tests under 4.02 and as a result `make all-supported-ocaml-versions` was doing nothing... I changed it build `@install` as well. I fixed the issue and made a 1.3.1 release.

Can't wait to stop supporting 4.02 though...
